packages/shorebird_cli/lib/src/commands/init_command.dart

```diff
@@ -92,6 +86,59 @@ If you want to reinitialize Shorebird, please run "shorebird init --force".''');
       return ExitCode.software.code;
     }
 
+    final shorebirdYaml = shorebirdEnv.getShorebirdYaml();
+    Set<String> newFlavors;
+    if (shorebirdYaml?.flavors != null) {
+      final existingFlavors = shorebirdYaml!.flavors!.keys.toSet();
+      newFlavors = productFlavors
+          .where((element) => !existingFlavors.contains(element))
+          .toSet();
+    } else {
+      newFlavors = {};
+    }
+
+    // New flavors not being empty means that we have existing flavors, which
+    // means that there is already an existing app.
+    // If the --force flag is present, we will completely reinit the app and
+    // don't care about which flavors are new.
+    if (!force && newFlavors.isNotEmpty) {
+      logger.info('New flavors detected: ${newFlavors.join(', ')}');
+      final updateShorebirdYamlProgress =
+          logger.progress('Adding flavors to shorebird.yaml');
+
+      final AppMetadata existingApp;
+      try {
+        existingApp =
+            await codePushClientWrapper.getApp(appId: shorebirdYaml!.appId);
+      } catch (e) {
+        updateShorebirdYamlProgress.fail('Failed to get existing app info: $e');
+        return ExitCode.software.code;
+      }
+
+      final deflavoredAppName =
+          existingApp.displayName.replaceAll(RegExp(r'\(.*\)'), '').trim();
+      final flavorsToAppIds = shorebirdYaml.flavors!;
+      for (final flavor in newFlavors) {
+        final app = await codePushClientWrapper.createApp(
+          appName: '$deflavoredAppName ($flavor)',
+        );
+        flavorsToAppIds[flavor] = app.id;
+      }
+      _addShorebirdYamlToProject(
+        shorebirdYaml.appId,
+        flavors: flavorsToAppIds,
+      );
+      updateShorebirdYamlProgress.complete('Flavors added to shorebird.yaml');
+      return ExitCode.success.code;
+    }
+
+    if (!force && shorebirdEnv.hasShorebirdYaml) {
+      logger.err('''
+A "shorebird.yaml" already exists.
+If you want to reinitialize Shorebird, please run "shorebird init --force".''');
```

Comment:
minor but might be nice to split the recommendation into a `.info` and wrap the command in `lightCyan` for consistency

Reply:
Done